native/libcst/src/parser/grammar.rs

```diff
@@ -0,0 +1,2930 @@
+                rhs:(eq:lit("=") d:annotated_rhs() {(eq, d)})? {
+                    SmallStatement::AnnAssign(make_ann_assignment(a, col, ann, rhs))
+            }
+            / lhs:(t:star_targets() eq:lit("=") {(t, eq)})+ rhs:(yield_expr() / star_expressions()) !lit("=") {
+                SmallStatement::Assign(make_assignment(lhs, rhs))
+            }
+            / t:single_target() op:augassign() rhs:(yield_expr() / star_expressions()) {
+                SmallStatement::AugAssign(make_aug_assign(t, op, rhs))
+            }
+
+        rule augassign() -> AugOp<'a>
+            = &(lit("+=") / lit("-=") / lit("*=") / lit("@=") /  lit("/=") / lit("%=") / lit("&=") / lit("|=") / lit("^=") / lit("<<=")
+                / lit(">>=") / lit("**=") / lit("//=")) tok:_ {?
+                    make_aug_op(tok).map_err(|_| "aug_op")
+            }
+
+        rule class_def() -> ClassDef<'a>
+            = d:decorators() c:class_def_raw() { c.with_decorators(d) }
+            / class_def_raw()
+
+        rule class_def_raw() -> ClassDef<'a>
+            = kw:lit("class") n:name() arg:(l:lpar() a:arguments()? r:rpar() {(l, a, r)})?
+                col:lit(":") b:block() {?
+                    make_class_def(kw, n, arg, col, b)
+            }
+
+        #[cache]
+        rule block() -> Suite<'a>
+            = n:tok(NL, "NEWLINE") ind:tok(Indent, "INDENT") s:statements() ded:tok(Dedent, "DEDENT") {
+                make_indented_block(n, ind, s, ded)
+            }
+            / s:simple_stmt() {
+                make_simple_statement_suite(s)
+            }
+
+        rule star_expressions() -> Expression<'a>
+            = first:star_expression()
+                rest:(comma:comma() e:star_expression() { (comma, expr_to_element(e)) })+
+                comma:comma()? {
+                    Expression::Tuple(make_tuple(expr_to_element(first), rest, comma, None, None))
+            }
+            / e:star_expression() comma:comma() {
+                Expression::Tuple(make_tuple(expr_to_element(e), vec![], Some(comma), None, None))
+            }
+            / star_expression()
+
+        #[cache]
+        rule star_expression() -> Expression<'a>
+            = star:lit("*") e:bitwise_or() {
+                Expression::StarredElement(make_starred_element(star, expr_to_element(e)))
+            }
+            / expression()
+
+        rule star_named_expressions() -> Vec<Element<'a>>
+            = exps:separated_trailer(<star_named_expression()>, <comma()>) {
+                    comma_separate(exps.0, exps.1, exps.2)
+            }
+
+        rule star_named_expression() -> Element<'a>
+            = star:lit("*") e:bitwise_or() {
+                Element::Starred(make_starred_element(star, expr_to_element(e)))
+            }
+            / e:named_expression() { expr_to_element(e) }
+
+        #[cache]
+        rule expression() -> Expression<'a>
+            = _conditional_expression()
```

Comment:
I pulled out `_conditional_expression` because it's also used for parsing f-strings. The official grammar doesn't include a rule for f-strings, they're parsed in a fairly ad-hoc way. Arguably this implementation is "more correct" for some definition of correct.